airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java

```diff
@@ -197,31 +206,55 @@ private void produceMessages(final AutoCloseableIterator<AirbyteMessage> message
   }
 
   @VisibleForTesting
-  static void consumeWriteStream(final AirbyteMessageConsumer consumer) throws Exception {
-    // use a Scanner that only processes new line characters to strictly abide with the
-    // https://jsonlines.org/ standard
-    final Scanner input = new Scanner(System.in, StandardCharsets.UTF_8).useDelimiter("[\r\n]+");
-    consumer.start();
-    while (input.hasNext()) {
-      consumeMessage(consumer, input.next());
+  static void consumeWriteStream(final SerializedAirbyteMessageConsumer consumer) throws Exception {
+    try (final BufferedInputStream bis = new BufferedInputStream(System.in);
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
+      consumeWriteStream(consumer, bis, baos);
     }
   }
 
-  private static void runConsumer(final AirbyteMessageConsumer consumer) throws Exception {
-    watchForOrphanThreads(
-        () -> consumeWriteStream(consumer),
-        () -> System.exit(FORCED_EXIT_CODE),
-        INTERRUPT_THREAD_DELAY_MINUTES,
-        TimeUnit.MINUTES,
-        EXIT_THREAD_DELAY_MINUTES,
-        TimeUnit.MINUTES);
+  @VisibleForTesting
+  static void consumeWriteStream(final SerializedAirbyteMessageConsumer consumer,
+                                 final BufferedInputStream bis,
+                                 final ByteArrayOutputStream baos)
+      throws Exception {
+    consumer.start();
+
+    final byte[] buffer = new byte[8192]; // 8K buffer
```

Comment:
any specific reason to use 8k?

Reply:
is this from somewhere else? looks familiar.

Reply:
8k bytes is the default buffer size for `BufferedInputStream`

Reply:
+1 to what ryan said.